### PR TITLE
Feat: Update API Gateway Configuration

### PR DESCRIPTION
**Pull request by williamputraintan**
- Group the APIGW microservice config as one props across microservice.
- Integrate [orcaui's cognito](https://github.com/umccr/infrastructure/pull/472) to microservice APIGW
- Update some CORS config on APIGW

### CODE REVIEW

File: lib/workload/stateless/stacks/filemanager/deploy/stack.ts [hunk 1]
```diff
@@ -141,20 +137,21 @@ export class Filemanager extends Stack {
...
 
     const apiIntegration = new HttpLambdaIntegration('ApiIntegration', apiLambda.function);
 
-    new HttpRoute(this, 'HttpRoute', {
+    new HttpRoute(this, 'GetHttpRoute', {
+      httpApi: httpApi,
+      integration: apiIntegration,
+      routeKey: HttpRouteKey.with('/{proxy+}', HttpMethod.GET),
+    });
+
```
Comment (williamputraintan) [inline lib/workload/stateless/stacks/filemanager/deploy/stack.ts:150]: @mmalenic Wanted to confirm from the [FM docs](https://github.com/umccr/orcabus/blob/main/lib/workload/stateless/stacks/filemanager/docs/API_GUIDE.md), FM only used GET and PATCH method?

**Comment by williamputraintan** [comment]
Apparently, APIGW JWT Authorizer works best where we define each method instead of `ANY`. It seems that defining this way allows the `OPTIONS` without an `authorizer` header included, whereas ANY require all methods to have an `auth` header.

File: skel/django-api/deploy/stack.ts [hunk 2]
```diff
@@ -116,20 +112,15 @@ export class ProjectNameStack extends Stack {  // FIXME change construct name
...
-      apiName: 'SequenceRunManager',
-      customDomainNamePrefix: 'sequence',
-      ...props,
-    });
+    const srmApi = new ApiGatewayConstruct(this, 'ApiGateway', props.apiGatewayCognitoProps);
     const httpApi = srmApi.httpApi;
 
     const apiIntegration = new HttpLambdaIntegration('ApiIntegration', apiFn);
 
-    new HttpRoute(this, 'HttpRoute', {
```
Comment (williamputraintan) [inline skel/django-api/deploy/stack.ts:129]: @victorskl @reisingerf On the SRM and WFM only the `GET` method currently used?

Comment (victorskl) [reply skel/django-api/deploy/stack.ts:129]: Read only `GET` for now.

**Review by victorskl — APPROVED** [review]

**Review by reisingerf — APPROVED** [review]

File: lib/workload/stateless/stacks/filemanager/deploy/stack.ts (hunk 1 above)
Comment (mmalenic) [reply lib/workload/stateless/stacks/filemanager/deploy/stack.ts:150]: There's a `POST` route as well on `/api/v1/ingest` which manually ingests data from an SQS queue. It isn't being used right now, but could you also please add it here?

**Review by mmalenic — APPROVED** [review]

File: lib/workload/stateless/stacks/filemanager/deploy/stack.ts (hunk 1 above)
Comment (williamputraintan) [reply lib/workload/stateless/stacks/filemanager/deploy/stack.ts:152]: Resolved - ccf38b2de940bba17076993616dade51fa375d78

**Pull request merged**